XLS-0081d-permissioned-dex/README.md

```diff
@@ -0,0 +1,593 @@
+
+* The first page (also called the root) of an Owner or NFT Offer directory
+* The first page of an Offer directory
+* Later pages of either type
+
+The first page of an Offer directory has a special ID: the higher 192 bits define the order book, and the remaining 64 bits define the exchange rate of the offers in that directory. (The ID is big-endian, so the book is in the more significant bits, which come first, and the quality is in the less significant bits which come last.) This provides a way to iterate through an order book from best offers to worst. Specifically: the first 192 bits are the first 192 bits of the [SHA-512Half](https://xrpl.org/docs/references/protocol/data-types/basic-data-types#hashes) of the following values, concatenated in order:
+
+* The Book directory space key (`0x0042`)
+* The 160-bit currency code from the `TakerPaysCurrency`
+* The 160-bit currency code from the `TakerGetsCurrency`
+* The AccountID from the `TakerPaysIssuer`
+* The AccountID from the `TakerGetsIssuer`
+* **The `DomainID` of the orderbook, if applicable**
+
+The lower 64 bits of an Offer directory's ID represent the `TakerPays` amount divided by `TakerGets` amount from the offer(s) in that directory as a 64-bit number in the XRP Ledger's internal amount format.
+
+If the DirectoryNode is not the first page in the directory, it has an ID that is the [SHA-512Half](https://xrpl.org/docs/references/protocol/data-types/basic-data-types#hashes) of the following values, concatenated in order:
+
+* The DirectoryNode space key (`0x0064`)
+* The ID of the root DirectoryNode
+* The page number of this object. (Since 0 is the root DirectoryNode, this value is an integer 1 or higher.)
+
+## 4. Transaction: `OfferCreate`
+
+The `OfferCreate` transaction creates an offer on the CLOB DEX. This transaction type already exists on the XRPL, but is being extended as a part of this spec to also support permissioned DEX domains.
+
+### 4.1. Fields
+
+<details>
+<summary>
+
+As a reference, [here](https://xrpl.org/docs/references/protocol/transactions/types/offercreate) are the existing fields for the `OfferCreate` transaction.
+</summary>
+
+| Field Name | Required? | JSON Type | Internal Type | Description |
+|------------|-----------|-----------|---------------|-------------|
+|`Expiration`| |`number`|`UInt32`|Time after which the offer is no longer active, in seconds since the Ripple Epoch.|
+|`OfferSequence`| |`number`|`UInt32`|An offer to delete first, specified in the same way as `OfferCancel`.|
+|`TakerGets`|✔️|`string`|`Amount`|The amount and type of currency being sold.|
+|`TakerPays`|✔️|`string`|`Amount`|The amount and type of currency being bought.|
+</details>
+
+We propose these additions:
+
+| Field Name | Required? | JSON Type | Internal Type | Description |
+|------------|-----------|-----------|---------------|-------------|
+|`DomainID`| |`string`|`Hash256`|The domain that the offer must be a part of.|
```

Comment:
what kind of validations do we need to allow someone create an offer with a domainID? can everyone create it?

Reply:
Only domain members can create it.